Consistify trailing punctuation in spec

janekdb opened the pull request:
Three types of change,

 - Drop trailing comma on some lists
 - Add missed commas on some lists
 - Drop `  .` on some examples
 - Correct =/== on linearization examples

I have assumed trailing `  .` in `scala` examples was not significant - I couldn't infer any meaning in it but happy to be corrected.

## TODO
 - [x] Address linearization comments
 - [x] Address comments

janekdb commented:
The rich diff option avoids the puzzling blood-bath of red on the first file,
![image](https://cloud.githubusercontent.com/assets/1123855/24244888/a3852222-0fb8-11e7-9ef9-09a85532e49d.png)

l0rinc commented on spec/06-expressions.md:
````diff
@@ -224,7 +224,7 @@ Then we have:
 ```scala
 (new A).superA == "Root",
````
Please join these lines also :)

l0rinc commented on spec/07-implicits.md:
````diff
@@ -218,7 +218,7 @@ which implicit arguments are searched is
 ```scala
 List[List[Int]] => Ordered[List[List[Int]]],
-List[Int] => Ordered[List[Int]]
+List[Int] => Ordered[List[Int]],
````
the list seems to continue here with `Int =>`

janekdb replied:
Yes, so adding the missing comma makes sense here don't you think?

l0rinc replied:
My mistake, yes :)
:+1:

l0rinc commented on spec/07-implicits.md:
```diff
@@ -239,7 +239,7 @@ of types for which implicit arguments are searched is
 Throwable => Ordered[Throwable],
-Throwable => Ordered[Throwable],
+Throwable => Ordered[Throwable]
 ...
```
if we interpret `...` as more elements to follow, we should keep the `,`.
Otherwise you could delete that also

janekdb replied:
Agreed. Restored.

l0rinc commented on spec/03-types.md:
````diff
@@ -507,7 +507,7 @@ Assume the class definitions
 
 ```scala
 class Ref[T]
-abstract class Outer { type T } .
````
:+1:, why was there a trailing `.` in the first place?

janekdb commented:
@paplorinc Ready for review.

janekdb committed:
Fix == in spec linearization examples

Also

 - Consistify trailing punctuation
 - Use whitespace to group linearization examples

l0rinc commented on spec/06-expressions.md:
```diff
@@ -222,9 +222,14 @@ the linearization of class `D` is `{D, B, A, Root}`.
-(new A).superA == "Root",
-                          (new C).superB = "Root", (new C).superC = "B",
-(new D).superA == "Root", (new D).superB = "A",    (new D).superD = "B",
+(new A).superA == "Root"
```
:+1:

SethTisue commented:
thank you!
